File: src/model/Atleta.java
package model;

import java.time.LocalDate;

public class Atleta extends Pessoa {
    //private String posicao;
    private String nCamisa;
    private double valorMercado;
    private LocalDate dataContratacao;
    private LocalDate fimContrato;
    private Stats stats;
    private Posicao posicao;



    public double getValorMercado() {
        return valorMercado;
    }

    public void setValorMercado(double valorMercado) {
        this.valorMercado = valorMercado;
    }

    public String getnCamisa() {
        return nCamisa;
    }

    public void setnCamisa(String nCamisa) {
        this.nCamisa = nCamisa;
    }

    public LocalDate getFimContrato() {
        return fimContrato;
    }

    public void setFimContrato(LocalDate fimContrato) {
        this.fimContrato = fimContrato;
    }

    public LocalDate getDataContratacao() {
        return dataContratacao;
    }

    public void setDataContratacao(LocalDate dataContratacao) {
        this.dataContratacao = dataContratacao;
    }

    public Posicao getPosicao() {
        return posicao;
    }

    public Stats getStats() {
        return stats;
    }

    public void setStats(Stats stats) {
        this.stats = stats;
    }

    public void setPosicao(Posicao posicao) {
        this.posicao = posicao;
    }

    @Override
    public void exibirDados() {
        System.out.println("Nome: " + nome);
        System.out.println("Idade: " + idade);
        System.out.println("Posição: " + posicao);
        System.out.println("Numero Camisa: " + nCamisa);
        System.out.println("Valor de mercado" + valorMercado);
        System.out.println("Data de Contratacao: " + dataContratacao);
        System.out.println("Fim Contrato: " + fimContrato);
        System.out.println("Stats: " + stats);
    }

    @Override
    public String toString() {
        return "Nome: " + nome + ", Posição: " + posicao;
    }
}

File: sgfc_too/lib/Atleta.dart
import '../Pessoa.dart';
import '../Stats.dart';
import '../Posicao.dart';

class Atleta extends Pessoa {
  String _nCamisa;
  double _valorMercado;
  DateTime _dataContratacao;
  DateTime _fimContrato;
  Stats _sstats;
  Posicao _pposicao;
  double _salario;

  Atleta({
    required String nome,
    required String sobrenome,
    required DateTime dataNascimento,
    required String nCamisa,
    required double valorMercado,
    required DateTime dataContratacao,
    required DateTime fimContrato,
    required Stats sstats,
    required Posicao pposicao,
    required double salario,
  })  : _nCamisa = nCamisa,
        _valorMercado = valorMercado,
        _dataContratacao = dataContratacao,
        _fimContrato = fimContrato,
        _sstats = sstats,
        _pposicao = pposicao,
        _salario = salario,
        super(
          nome: nome,
          sobrenome: sobrenome,
          dataNascimento: dataNascimento,
        );

  // Getters e Setters
  String get nCamisa => _nCamisa;
  set nCamisa(String value) => _nCamisa = value;

  double get valorMercado => _valorMercado;
  set valorMercado(double value) => _valorMercado = value;

  DateTime get dataContratacao => _dataContratacao;
  set dataContratacao(DateTime value) => _dataContratacao = value;

  DateTime get fimContrato => _fimContrato;
  set fimContrato(DateTime value) => _fimContrato = value;

  Stats get stats => _sstats;
  set stats(Stats value) => _sstats = value;

  Posicao get posicao => _pposicao;
  set posicao(Posicao value) => _pposicao = value;

  double get salario => _salario;
  set salario(double value) => _salario = value;

  @override
  void bonusSalario() {
    if (_pposicao == Posicao.Atacante && _sstats.gols > 50 && _sstats.gols < 100) {
      _salario += _salario * 0.3;
    } else if (_pposicao == Posicao.Atacante && _sstats.gols > 100) {
      _salario += _salario * 0.6;
    }
  }

  @override
  void exibirDados() {
    print('Nome: $nome');
    print('Idade: ${calcularIdade()}');
    print('Posição: $_pposicao');
    print('Número da Camisa: $_nCamisa');
    print('Valor de Mercado: $_valorMercado');
    print('Data de Contratação: $_dataContratacao');
    print('Fim do Contrato: $_fimContrato');
    print('Stats: $_sstats');
  }

  @override
  String toString() {
    return 'Nome: $nome, Posição: $_pposicao';
  }
}
